lib/sparse/src/index/inverted_index/inverted_index_mmap.rs

```diff
@@ -42,18 +42,26 @@ struct PostingListFileHeader {
 }
 
 impl InvertedIndex for InvertedIndexMmap {
-    fn open(path: &Path) -> std::io::Result<Option<Self>> {
-        Self::load(path).map(Some)
+    fn open(path: &Path) -> std::io::Result<Self> {
+        Self::load(path)
+    }
+
+    fn save(&self, path: &Path) -> std::io::Result<()> {
+        if path != self.path {
+            // TODO(sparse): do we need to save index to another direction?
+            todo!("Cannot save to a different path")
```

Comment:
can we have a debug assert instead and log?

Reply:
changed into debug assert